org.eclipse.scanning.api/src/org/eclipse/scanning/api/points/AbstractPosition.java

```diff
@@ -96,9 +100,11 @@ public final boolean equals(Object obj, boolean checkStep) {
 				return false;
 		}
 
-		final Collection<String> ours   = getNames();
-		final Collection<String> theirs = ((IPosition)obj).getNames();
-		if (!equals(ours, theirs)) return false;		
+		final List<String> ours = new ArrayList<String>(getNames());
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Replace the type specification in this constructor call with the diamond operator ("<>"). (sonar.java.source not set. Assuming 7 or greater.) [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2293)